brave/brave-core
# Add sidebar enabled experiment metrics

Resolves https://github.com/brave/brave-browser/issues/36815



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

For "A" metrics, test using `--enable-features="SidebarShowAlwaysOnStable:open_one_shot_leo_panel/false"` switch. For "B" metrics, test using `--enable-features="SidebarShowAlwaysOnStable:open_one_shot_leo_panel/true"` switch.

For `Brave.Core.SidebarSettings.SidebarEnabledA/B`:

0. Start with fresh profile, ensure one metric only exists if using one of the above switches, with the value 0.
1. Change the sidebar setting, ensure metric updates accordingly.

For `Brave.AIChat.UsageWeekly.SidebarEnabledA/B`, `Brave.AIChat.UsageDaily.SidebarEnabledA/B` and `Brave.AIChat.Enabled.SidebarEnabledA/B`:

0. Start with fresh profile with no switches, ensure no metric similar to `Brave.AIChat.UsageWeekly` or `Brave.AIChat.UsageDaily.2` or `Brave.AIChat.Enabled.2` exists.
1. Use Leo, ensure `Brave.AIChat.UsageWeekly`, `Brave.AIChat.UsageDaily.2` and `Brave.AIChat.Enabled.2` exists with the correct value.
2. Start with fresh profile using one of the switches, ensure no relevant metrics exist.
3. Use Leo, ensure  `Brave.AIChat.UsageWeekly.SidebarEnabledA/B`, `Brave.AIChat.UsageDaily.SidebarEnabledA/B`, and `Brave.AIChat.Enabled.SidebarEnabledA/B`  exists with the correct value.
